arcgis-ios-sdk-samples/Maps/Generate offline map (overrides)/OfflineMapParameterOverridesViewController.swift

```diff
@@ -0,0 +1,263 @@
+    
+    //MARK: - Outlets
+
+    /// The min scale level for the output. Note that lower values are zoomed further out,
+    /// i.e. 0 has the least detail, but one tile covers the entire Earth.
+    @IBOutlet weak var minScaleLevelSlider: UISlider!
+    
+    /// The max scale level for the output. Note that higher values are zoomed further in,
+    /// i.e. 23 has the most detail, but each tile covers a tiny area.
+    @IBOutlet weak var maxScaleLevelSlider: UISlider!
+    
+    /// The extra padding added to the extent rect to fetch a larger area, in meters.
+    @IBOutlet weak var basemapExtentBufferSlider: UISlider!
+    
+    /// Switch indicating if the system valves layer should be included in the download.
+    @IBOutlet weak var includeSystemValvesSwitch: UISwitch!
+    ///Switch indicating if the service connections layer should be included in the download.
+    @IBOutlet weak var includeServiceConnectionsSwitch: UISwitch!
+    
+    /// The minimum flow rate by which to filter features in the Hydrants layer, in gallons per minute.
+    @IBOutlet weak var minHydrantFlowRateSlider: UISlider!
+    
+    ///Switch indicating if the pipe layers should be restricted to the extent frame.
+    @IBOutlet weak var cropWaterPipesToExtentSwitch: UISwitch!
+    
+    //MARK: - Actions
+    
+    @IBAction func sliderChangeAction(_ sender: UISlider) {
+        if sender == minScaleLevelSlider {
+            // Disallow a min value greater than the maximum
+            if sender.value > maxScaleLevelSlider.value {
+                sender.value = maxScaleLevelSlider.value
+            }
+        }
+        else if sender == maxScaleLevelSlider {
+            // Disallow a max value less than the minimum
+            if sender.value < minScaleLevelSlider.value {
+                sender.value = minScaleLevelSlider.value
+            }
+        }
+        // Update this slider's text field to display the new value
+        updateTextField(for: sender)
+    }
+    
+    //MARK: - Text field updating
+    
+    private func updateTextField(for slider: UISlider){
+        let numberFormatter = NumberFormatter()
+        if let text = numberFormatter.string(from: slider.value as NSNumber),
+            let superview = slider.superview,
+            let field = superview.subviews.first(where: { $0 is UITextField }) as? UITextField{
+            field.text = text
+        }
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // Update each text field to display its slider's default value
+        for slider in [minScaleLevelSlider,
+                       maxScaleLevelSlider,
+                       basemapExtentBufferSlider,
+                       minHydrantFlowRateSlider] {
+            updateTextField(for: slider!)
+        }
+    }
+    
+    //MARK: - Cancelling
+    
+    /// The completion handler to run if the user clicks cancel
+    var cancelHandler: ((OfflineMapParameterOverridesViewController)->())?
```

Comment:
It used to be ambiguous whether to use `()` or `Void` for a function that doesn't return anything, but Swift finally settled on using [void](https://developer.apple.com/documentation/swift/void).

Reply:
Oh gotcha, I knew they were equivalent but I didn't know `Void` was recommend for closure types.